docs/docs/data-structures/tree/schema-evolution/index.mdx

```diff
@@ -0,0 +1,219 @@
+SharedTree will throw an assert if the application attempts to access the document's data with an incompatible view schema.
+
+Initially, a document is always compatible with the application that created it (i.e. the view schema is compatible with the stored schema).
+However, later on the application might change the view schema.
+Such a change might be backwards compatible, forwards compatible, or neither or both.
+
+### Understanding Compatibility
+
+#### Backwards Compatibility
+
+A change to the view schema is **backwards compatible** if the *new view schema* is compatible with the *old stored schema*.
+In practice, this means that the new version of the application will be able to [upgrade](./index.mdx#schema-upgrade-process) documents to the new schema (note that the process of upgrading might affect forwards compatibility - see below).
+
+If a change is made that is not backwards compatible, then existing documents will "break", that is, they can no longer be loaded by the application.
+
+#### Forwards Compatibility
+
+A change to the view schema is **forwards compatible** if an *old view schema* is compatible with the *new stored schema* that results from the upgrade.
+In practice, the older version of the application must be able to "view" the new document after it has been upgraded.
+
+If a change is made that is backwards compatible, but not forwards compatible, then old versions of the application will "break" when trying to load newer/upgraded documents.
+To prevent this, the application can employ a [staged rollout](./index.mdx#staged-rollouts).
+
+### Schema Change Examples
+
+#### 😎 Chill
+| Backwards Compatible | Forwards Compatible | Rollout Process |
+|-----------------------|----------------------|-----------------|
+| ✅                    | ✅                   | Normal          |
+
+**Examples:**
+- Adding an optional field when compatibility flag is enabled
+
+#### 🌶️ Spicy
+| Backwards Compatible | Forwards Compatible | Rollout Process |
+|-----------------------|----------------------|-----------------|
+| ✅                    | ❌                   | ⚠️ Staged       |
+
+**Examples:**
+- Adding a new allowed type (TODO link to how to doc)
+
+#### 👮 Go to Jail
+| Backwards Compatible | Forwards Compatible | Rollout Process |
+|-----------------------|----------------------|-----------------|
+| ❌                    | ❌                   | 🚫 Forbidden    |
+
+**Examples:**
+- Adding/removing a required field
+- Removing an allowed type from a field
+- Renaming a node identifier
+
+
+### Checking Compatibility
+
+Use [`TreeView.compatibility`](../../../api/fluid-framework/treeview-interface#compatibility-property) to check if your schema changes are compatible with existing documents:
+
+```typescript
+const view = tree.viewWith(configuration);
+const { compatibility } = view;
+
+if (!compatibility.isEquivalent) {
```

Comment:
I think we should just stop advertising `isEquivalent`. It doesn't really do anything sensible. It's safe to call `upgradeSchema()` even if nothing has changed - we will no-op and no harm done.